src/composition/supergraph/config/resolver/mod.rs

```diff
@@ -117,8 +125,17 @@ impl SupergraphConfigResolver<state::LoadRemoteSubgraphs> {
                 .call(FetchRemoteSubgraphsRequest::new(graph_ref.clone()))
                 .await
                 .map_err(|err| {
-                    LoadRemoteSubgraphsError::FetchRemoteSubgraphsError(Box::new(err))
+                    //the trait bound `rover_client::RoverClientError: From<<S as MakeService<(), FetchRemoteSubgraphsRequest>>::Error>` is not satisfied
+                    //the trait `From<<S as MakeService<(), FetchRemoteSubgraphsRequest>>::Error>` is not implemented for `rover_client::RoverClientError`
+
+                    match RoverClientError::from(err) {
+                        RoverClientError::PermissionError { msg } => {
+                            LoadRemoteSubgraphsError::FetchRemoteSubgraphsAuthError(Box::new(err))
+                        },
+                        _ =>  { LoadRemoteSubgraphsError::FetchRemoteSubgraphsError(Box::new(err)) },
+                    }
```

Comment:
I'm back in the same dang hole here. Even though this appears to me to literally be a `RoverClientError` (as that is the response type from the adjusted method in FetchAllSubgraphsService), here we don't know that, and I need some way to negotiate the type conversion. I banged my shins on writing a From in the error file, but didn't get anything working over there.

Appreciate any feedback or pointers, even if it's "This is all wrong" :)